cpp/src/utilities/cython.cpp

```diff
@@ -140,155 +265,51 @@ void populate_graph_container(graph_container_t& graph_container,
-        std::unique_ptr<experimental::graph_view_t<int, int, double, false, false>>(g);
-      graph_container.graph_ptr_type = graphTypeEnum::graph_view_t_double;
-    }
-  }
 }
 
-void graph_container_t::get_vertex_identifiers(void* c_identifier)
-{
-  CUGRAPH_EXPECTS(graph_ptr_type != graphTypeEnum::null,
-                  "get_vertex_identifiers() cannot be called on an uninitialized container");
-
-  switch (graph_ptr_type) {
-    case graphTypeEnum::GraphCSRViewFloat: {
-      graph_ptr_union.GraphCSRViewFloatPtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    case graphTypeEnum::GraphCSCViewFloat: {
-      graph_ptr_union.GraphCSCViewFloatPtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    case graphTypeEnum::GraphCOOViewFloat: {
-      graph_ptr_union.GraphCOOViewFloatPtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    case graphTypeEnum::GraphCSRViewDouble: {
-      graph_ptr_union.GraphCSRViewDoublePtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    case graphTypeEnum::GraphCSCViewDouble: {
-      graph_ptr_union.GraphCSCViewDoublePtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    case graphTypeEnum::GraphCOOViewDouble: {
-      graph_ptr_union.GraphCOOViewDoublePtr->get_vertex_identifiers(
-        static_cast<int32_t*>(c_identifier));
-    } break;
-    default: {
-      CUGRAPH_FAIL("unexpected weight type");
-    }
-  }
-}
+
+////////////////////////////////////////////////////////////////////////////////
 
 // Wrapper for calling Louvain using a graph container
 template <typename weight_t>
 std::pair<size_t, weight_t> call_louvain(raft::handle_t const& handle,
                                          graph_container_t const& graph_container,
+                                         void* identifiers,
                                          void* parts,
                                          size_t max_level,
                                          weight_t resolution)
 {
-  std::pair<size_t, float> results;
+  std::pair<size_t, weight_t> results;
 
   // FIXME: the only graph types currently in the container have ints for
   // vertex_t and edge_t types. In the future, additional types for vertices and
   // edges will be available, and when that happens, additional castings will be
   // needed for the 'parts' arg in particular. For now, it is hardcoded to int.
-  if (graph_container.graph_ptr_type == graphTypeEnum::GraphCSRViewFloat) {
+  if (graph_container.graph_ptr_type == graphTypeEnum::graph_t_float_mg) {
+    results = louvain(handle,
+                      graph_container.graph_ptr_union.graph_t_float_mg_ptr->view(),
+                      reinterpret_cast<int*>(parts),
+                      max_level,
+                      static_cast<float>(resolution));
+
```

Comment:
We'll need some kind of graph_t or graph_view_t analog to get_vertex_identifiers...

Reply:
You can retrieve vertex ranges with get_local_vertex_first() & get_local_vertex_last().

Then, can't you just use thrust::make_counting_iterator()? This will save both memory & computing than explicitly creating an array and filling the array with thrust::sequence.

Reply:
The python code is constructing a data frame with a column called 'vertex' that contains these identifiers that is returned to the caller.

I think something like:

`thrust::copy(..., thrust::make_counting_iterator(get_local_vertex_first()), thrust::make_counting_iterator(get_local_vertex_last(), identifiers)`

is probably the right implementation.

Reply:
I'll defer this to the next PR for this code.

Reply:
Created PR #1178

Reply:
@ChuckHastings is planning on addressing this in (I think) PR #1172